ocs_ci/helpers/helpers.py

```diff
@@ -348,7 +350,13 @@ def create_pod(
             pod_data["spec"]["containers"][0]["volumeDevices"][0]["name"] = (
                 pod_data.get("spec").get("volumes")[0].get("name")
             )
-
+    if security_context:
```

Comment:
Since you are adding the sc in the dc yaml, you dont need to create a new simple-app yaml again.
you could use the existing fedora deployment config and update the sc details